web/src/beta/components/RadioGroup/index.tsx

```diff
@@ -4,50 +4,49 @@ import RadioBox from "@reearth/beta/components/RadioGroup/RadioBox";
 import { styled } from "@reearth/services/theme";
 
 export type Option = {
-  key: string;
-  value: string;
-  selected: boolean;
+  label?: string;
+  keyValue: string;
 };
 
 export type RadioGroupProps = {
   options: Option[];
   layout?: "vertical" | "horizontal";
+  selectedValue?: string;
   onChange?: (value: string) => void;
 };
 
-const RadioGroup: React.FC<RadioGroupProps> = ({ options, layout, onChange }) => {
-  const [currentOptions, updateOptions] = useState<Option[]>(options);
+const RadioGroup: React.FC<RadioGroupProps> = ({ options, layout, selectedValue, onChange }) => {
   const [key, setKey] = useState(0);
 
   const handleRadioChange = useCallback(
     (value: string) => {
-      updateOptions(
-        currentOptions.map(option => ({
-          ...option,
-          selected: !option.selected && option.value === value,
-        })),
-      );
+      if (value === selectedValue) return;
+
       setKey(prevKey => prevKey + 1);
       onChange?.(value);
     },
-    [currentOptions, onChange],
+    [onChange, selectedValue],
   );
+
   return (
     <RadioGroupContainer layout={layout}>
-      {currentOptions.map(option => (
+      {options.map((option, index) => (
         <RadioBox
-          key={`${option.key}-${key}`}
-          value={option.value}
-          selected={option.selected}
-          onClick={() => handleRadioChange(option.value)}
+          key={`${option.keyValue}-${key}-${index}`}
+          keyValue={option.keyValue}
+          selected={option.keyValue === selectedValue}
+          label={option.label || option.keyValue}
```

Comment:
We don't want label to ever be keyValue. The keyValue will be stuff like "url", "csv", "default_tile", etc. We don't want to display that. We will want label to be required 🙏🏻

Reply:
sorry sorry I forget to remove it ,sorry